api/src/services/listing.service.ts

```diff
@@ -190,8 +190,9 @@ export class ListingService implements OnModuleInit {
 
     // if passed in page and limit would result in no results because there aren't that many listings
     // revert back to the first page
+
     let page = params.page;
-    if (count && params.limit && params.limit !== 'all' && params.page > 1) {
+    if (count && params.limit && params.limit !== -1 && params.page > 1) {
```

Comment:
Since we allow it to be any number, should we be checking `>0` instead of `!== -1`. Could allow for errors if api is hit incorrectly as is